src/VecSim/algorithms/hnsw/hnswlib.h

```diff
@@ -110,8 +112,15 @@ class HierarchicalNSW : public VecsimBaseObject {
                           size_t Mcurmax, tableint *node_neighbors,
                           const vecsim_stl::set<tableint> &orig_neighbors, tableint *removed_links,
                           size_t *removed_links_num);
+    dist_t process_candidate(tableint curNodeId, const void *data_point, size_t layer, size_t ef,
```

Comment:
1. God idea
2. seems negligible, but I can try and profile the run
3. No, because there we want to save candidates as "extras" and not just popping them if we need to pop.